src/meta/src/stream/stream_manager.rs

```diff
@@ -263,80 +263,80 @@ impl GlobalStreamManager {
+                            let node_actors = table_fragments.worker_actor_ids();
+                            let cluster_info =
+                                self.cluster_manager.get_streaming_cluster_info().await;
+                            let node_actors = node_actors
+                                .into_iter()
+                                .map(|(id, actor_ids)| {
+                                    (
+                                        cluster_info.worker_nodes.get(&id).cloned().unwrap(),
+                                        actor_ids,
+                                    )
+                                })
+                                .collect_vec();
+                            let futures = node_actors.into_iter().map(|(node, actor_ids)| {
+                                let request_id = Uuid::new_v4().to_string();
+                                async move {
+                                    let client = self.env.stream_client_pool().get(&node).await?;
+                                    let request = DropActorsRequest {
+                                        request_id,
+                                        actor_ids,
+                                    };
+                                    client.drop_actors(request).await
+                                }
                             });
-                            return Err(MetaError::cancelled("create".into()));
+                            try_join_all(futures).await?;
+
+                            self.fragment_manager
+                                .drop_table_fragments_vec(&HashSet::from_iter(std::iter::once(
+                                    table_id,
+                                )))
+                                .await?;
+                        }
+                        if !table_fragments.is_created() {
+                            tracing::debug!(
+                                "cancelling streaming job {table_id} by issue cancel command."
+                            );
+                            self.barrier_scheduler
+                                .run_command(Command::CancelStreamingJob(table_fragments))
+                                .await?;
                         }
+                        let _ = finish_tx.send(()).inspect_err(|_| {
+                            tracing::warn!("failed to notify cancelled: {table_id}")
+                        });
+                        self.creating_job_info.delete_job(table_id).await;
+                        return Err(MetaError::cancelled("create".into()));
                     }
-                    CreatingState::Created => return Ok(()),
+                }
+                CreatingState::Created => {
+                    self.creating_job_info.delete_job(table_id).await;
+                    return Ok(());
                 }
             }
-        };
-
-        self.creating_job_info.delete_job(table_id).await;
-        res
+        }
+        Ok(())
```

Comment:
Bugfix 1.

Reply:
@yezizp2012 could you PTAL at this part? Previously the `try` block could return without calling `self.creating_job_info.delete_job(table_id).await`, whenever there was any update to the `CreatingState`.

This is because in `try blocks` the `return` will return in the function scope, rather than within the block scope: https://github.com/kwannoel/rust-experiments/blob/main/try-with-return.rs